core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -96,12 +103,38 @@ static ClientFactoryBuilder builder() {
      * Closes the default {@link ClientFactory}.
      */
     static void closeDefault() {
-        LoggerFactory.getLogger(ClientFactory.class).debug(
-                "Closing the default {}", ClientFactory.class.getSimpleName());
-        try {
-            DefaultClientFactory.DEFAULT.doClose();
-        } finally {
-            DefaultClientFactory.INSECURE.doClose();
+        final Logger logger = LoggerFactory.getLogger(ClientFactory.class);
+        logger.debug("Closing the default client factories");
+        final CompletableFuture<Void> closeFuture = CompletableFuture.allOf(
```

Comment:
Shouldn't we use `CompletableFutures.succesfulAsList(...)` (log the cause using `defaultValueMapper`) because we have to wait for all futures to be complete?